### PR TITLE
feat: API use a worker

### PR DESCRIPTION
**Pull request by jgorset**

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -89,14 +95,22 @@ def transcribe():
...
+
+            job = rq_queue.enqueue(
+                'transcriber.transcribe',
+                args=(filename,requestedModel,task,output,language),
+                result_ttl=3600*24*7,
+                on_success=mailer.send_success_email,
+                on_failure=mailer.send_failure_email
             )
 
+            return 'Bananese',201
```
Comment (Johannes-Andersen) [inline src/main.py:112]: Should this return any data? Would seem better to return a 201 response with the queue task ID maybe?

also the code below does not seem to be reachable

File: src/main.py [hunk 2]
```diff
@@ -113,8 +127,36 @@ def transcribe():
...
+    
+    if job.is_finished:
+        if output == "txt":
+            return result["text"]
+        if output == "json":
+            return result        
+        if output == "vtt":
+            return generate_vtt(result["segments"]), 200, {'Content-Type': 'text/vtt', 'Content-Disposition': 'attachment; filename=transcription.vtt'}
+        if output == "srt":
+            return generate_srt(result["segments"]), 200, {'Content-Type': 'text/plain', 'Content-Disposition': 'attachment; filename=transcription.srt'}
```
Comment (Johannes-Andersen) [inline src/main.py:147]: I may be missing something, but I dont see where `result` is defined? Does not look like it will work?

Comment (jgorset) [reply src/main.py:147]: I don't think you missed anything! I rather missed to remove this bit. Replacing with HTTP 201 and confirmation that it's added to the queue!

Comment (Johannes-Andersen) [reply src/main.py:136]: Hum… this is the download endpoint. Don’t we want this to return the data with a 200OK? We should definitely keep the result in my opinion

Comment (jgorset) [reply src/main.py:136]: We'll need that code again in the code to send e-mail, though, I think!

File: src/mailer.py [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+
+  # Disconnect from the server
+  smtp_server.quit()
 
 def send_success_email(job, connection, result, *args, **kwargs):
-    ssl_context = ssl.create_default_context()
+  email = job.meta.get('email')
+  subject = 'Your file is ready'
+
+  download_url = "http://127.0.0.1:3000/v1/download/" + job.id
```
Comment (Johannes-Andersen) [inline src/mailer.py:27]: This URL should come from an environment variable as this would work in production

File: src/mailer.py [hunk 4]
```diff
@@ -1,11 +1,38 @@
...
+  body = f'Your file is ready. Download it here: {download_url}'
```
Comment (Johannes-Andersen) [inline src/mailer.py:28]: The body should be multi line so they can download it with the different formats

http://127.0.0.1:3000/v1/download/abc123/?output=vtt
http://127.0.0.1:3000/v1/download/abc123/?output=srt
http://127.0.0.1:3000/v1/download/abc123/?output= json
http://127.0.0.1:3000/v1/download/abc123/?output=txt

File: src/mailer.py [hunk 5]
```diff
@@ -1,11 +1,38 @@
...
-        server.login('my@gmail.com', 'password')
-        server.sendmail('jojo@vg.no', email, 'Hello here is your transcript!')
+  send_mail(email, subject, body)
 
 def send_failure_email(job, connection, type, value, traceback):
-    pass
+  email = job.meta.get('email')
+  subject = 'The transcription failed'
+
+  body = 'Your file failed, were sorry, please try again :D'
```
Comment (Johannes-Andersen) [inline src/mailer.py:36]: We're sorry, but we were unable to transcribe the file you uploaded. Please try again or contact the team for assistance.

Ask Janni maybe for better messaging

**Pull request merged**